core/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -91,6 +94,36 @@ private[spark] abstract class YarnSchedulerBackend(
   }
 
   /**
+   * Override the DriverEndpoint to add extra logic for the case when an executor is disconnected.
```

Comment:
Can you make it explicit who this endpoint talks to? Otherwise it's kinda confusing to see two endpoints in the same class on the driver.
```
An [[RpcEndpoint]] that communicates with the executors and used internally within the driver.
Here we override the DriverEndpoint to ...
```